api/v1alpha1/constants.go

```diff
@@ -28,6 +28,8 @@ const (
 	ClusterAutoscalerVariableName = "clusterAutoscaler"
 	// ServiceLoadBalancerVariableName is the Service LoadBalancer config patch variable name.
 	ServiceLoadBalancerVariableName = "serviceLoadBalancer"
+	// RegistryMirrorVariableName is the OCI registry config patch variable name.
```

Comment:
we have `globalRegistryMirror` and now `registryMirror`. that could be confusing for end users. I cant of think of any better suggestion now. however can we take it offline and think about specific name that shows intention of this registry?

Reply:
Here is an example of what that would look like. When thinking of the name I intentionally based it on the existing field, but I can see now that it may be confusing since its not clear what if these should be set together or separately.

I do want to convey with the name that this registry is specifically designed to be a mirror and not just a standalone registry (that can be a different addon if we ever want to add that in the future).

```
    - name: clusterConfig
      value:
        addons:
          registryMirror: {}
          ....
        globalImageRegistryMirror:
          url: https://example.com
          credentials:
            secretRef:
              name: my-mirror-ca-cert
```

Reply:
>specifically designed to be a mirror 

can you please remind why we designed it to be a mirror?

Reply:
If we have two mirrors, registryDistribution mirror and global registry mirror, how containerd will work? 
Which mirror it would select to cache the images it cant find and pull it from the backend registry?

I believe we initially decided to use single global image registry mirror per cluster. Trying to recall reason behind that call. 🤔

Reply:
> can you please remind why we designed it to be a mirror?

Some of the images needed for the cluster will be baked into the OS, but most images will need to come from this internal registry.
Setting it as a mirror automatically (done in https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/1117) makes it so that users do really not need to understand the internals of how this registry is deployed and don't need to manually set it as a mirror.

Reply:
RE: multiple mirrors.

The single mirror limitation comes from just CAREN's API and IIRC it was done that way for simplicity.
But the [handler code](https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/blob/3640635d3b4f00fa177a9545c345e41351829241/pkg/handlers/generic/mutation/mirrors/templates/hosts.toml.gotmpl#L1) and [Containerd](https://github.com/containerd/containerd/blob/main/docs/hosts.md#setup-a-local-mirror-for-docker) both handle a list, Containerd will just go down the list trying each mirror until it can find the image, with the final one being the registry from the image name.

In this case CAREN will add this internal registry as the last mirror in the list.

Reply:
I proposed above to use `registry` as the addon name. I think that makes things clearer. Although we are expecting to use this as a registry mirror, from CAREN's perspective it could equally be used as a simple way to deploy a registry.

Reply:
Are we ok calling it `registry` and always configuring it as a `mirror` https://github.com/nutanix-cloud-native/cluster-api-runtime-extensions-nutanix/pull/1117?
Do we want to make that configurable?

That was the intent of calling it `registryMirror` since users would not have control if its a mirror or not.

Reply:
@supershal does `registry` sound good to you?